src/components/ChartContainer.js

```diff
@@ -73,6 +68,21 @@ const ChartContainer = forwardRef(
     const [dataURL, setDataURL] = useState("");
     const [download, setDownload] = useState("");
 
+    useEffect(() => {
+      updateScale(zoom);
+    }, [zoom]);
+
+    const updateScale = (zoom) => {
+      if (!transformComponentRef.current) {
+        return;
+      }
+
+      const { centerView } = transformComponentRef.current;
+      const { value } = zoom;
+
+      centerView(value, 300, 'easeOut')
```

Comment:
@jd-gray I think the animation is real nice, I wouldn't do no animation. You can send it to Andrew and see what he thinks regarding speed. 

I am curious about `centerView` however. Would this mean you can't pan a horizontal org chart and zoom? Thinking about our org chart now at Trainual, when zooming is it always going to try and snap your position to the center?

Reply:
Here is some behavior here: https://www.loom.com/share/4eb27b35a5fb4bdb8af758960656027e

As mentioned in https://www.loom.com/share/aac54e56ffd4452aa24cc36d15967c53 the dependency handlers:
- `zoomIn` / `zoomOut` was inaccurate due to the set scale amount, ie when increasing/decreasing the scale amount would be of - starts at 1 -> 1.51 -> 0.99 -> 1.52 ect
- `setTransform` kinda worked but I would have to calculate the x and y axis to zoom in on and I'm not sure how to do that

`centerView` seemed like the only thing that would work. I can give it another try and may end up redoing this either with another dependency or write my own code

Reply:
I may try out this https://github.com/d3/d3-zoom to see if it works better

Reply:
@jd-gray I think your first video looks great. 

I did watch your initial video, it was just a skinny vertically stacked chart so I wasn't sure what a wide chart would do. 

--

If you aren't happy w/your dependency you can try d3. But I think the video above looked perfect.

Reply:
If we wanted to have a zoom functionality that zooms into the center of where the org chart is 'looking', would that be difficult to implement?

> `setTransform` kinda worked but I would have to calculate the x and y axis to zoom in on and I'm not sure how to do that

From a mathematical perspective, it's easy, but I think it's about getting the numbers in order to do the math, isn't it? I noticed in the Loom video it seems to zoom to the right instead of straight on as well. Why is that? Is that because of the org chart being to the right on the screen a little bit (navigation bar on the left pushing it to the right, I mean..?? I doubt this is the case, but curious)

Reply:
@templeman15 @Tristan-Trainual I pushed up some updates to use the `setTransform` method, would you two be available to sync on this at some point? It doesn't have to be today but perhaps we can meet about it tomorrow?

Reply:
Yeah hit both or either of us up!

Reply:
I'd like to sync up and follow along! 💯